include/aero/actuator/ActuatorGenericTurbineSearchFunctor.h

```diff
@@ -0,0 +1,147 @@
+      dualNodalVolume_(stkBulk_.mesh_meta_data().template get_field<double>(
+        stk::topology::NODE_RANK, "dual_nodal_volume")),
+      innerLoopFunctor_(innerLoopFunctor)
+  {
+    actBulk_.coarseSearchElemIds_.sync_host();
+    actBulk_.coarseSearchPointIds_.sync_host();
+    innerLoopFunctor_.preloop();
+  }
+
+  // see ActuatorExecutorFASTSngp.C line 58
+  // index corresponds to turbines here
+  void operator()(int index) const
+  {
+
+    auto pointId = actBulk_.coarseSearchPointIds_.h_view(index);
+    auto elemId = actBulk_.coarseSearchElemIds_.h_view(index);
+
+    // get element topology 
+    const stk::mesh::Entity elem =
+      stkBulk_.get_entity(stk::topology::ELEMENT_RANK, elemId);
+    const stk::topology& elemTopo = stkBulk_.bucket(elem).topology();
+    MasterElement* meSCV =
+      MasterElementRepo::get_volume_master_element_on_host(elemTopo);
+
+    // element number of nodes and integration points
+    const unsigned numNodes = stkBulk_.num_nodes(elem);
+    const int numIp = meSCV->num_integration_points();
+
+    // just allocate for largest expected size (hex27)
+    STK_ThrowAssert(numIp <= 216);
+    STK_ThrowAssert(numNodes <= 27);
+
+    double scvip[216];
+    double elemcoords[27 * 3];
+    sierra::nalu::SharedMemView<double*> scvIp(&scvip[0], 216);
+    sierra::nalu::SharedMemView<double**> elemCoords(&elemcoords[0], 27, 3);
+
+    stk::mesh::Entity const* elem_nod_rels = stkBulk_.begin_nodes(elem);
+
+    // get element coordinates
+    for (unsigned i = 0; i < numNodes; i++) {
+      const double* coords =
+        stk::mesh::field_data(*coordinates_, elem_nod_rels[i]);
+      for (int j = 0; j < 3; j++) {
+        elemCoords(i, j) = coords[j];
+      }
+    }
+
+    meSCV->determinant(elemCoords, scvIp);
+
+    // relationship of element nodes to integration points
+    const auto* ipNodeMap = meSCV->ipNodeMap();
+
+    // loop over integration points
+    for (int nIp = 0; nIp < numIp; nIp++) {
+      const int nodeIndex = ipNodeMap[nIp];
+      stk::mesh::Entity node = elem_nod_rels[nodeIndex];
+      const double* nodeCoords = stk::mesh::field_data(*coordinates_, node);
+      const double dual_vol = *stk::mesh::field_data(*dualNodalVolume_, node);
+      double* sourceTerm = stk::mesh::field_data(*actuatorSource_, node);
+
+      // anything else that is required should be stashed on the functor
+      // during functor construction i.e. ActuatorBulk, flags, ActuatorMeta,
+      // etc.
+      //
+      // loop over actuator points. Don't need to change innerLoopFunctors
+      for (int actPtInd = 0; actPtInd < actBulk_.pointCentroid_.extent(0); actPtInd ++){
+        innerLoopFunctor_(actPtInd, nodeCoords, sourceTerm, dual_vol, scvIp[nIp]);
+      }
+  }
+  }
+
+  ActuatorBulk& actBulk_;
+  stk::mesh::BulkData& stkBulk_;
+  VectorFieldType* coordinates_;
+  VectorFieldType* actuatorSource_;
+  ScalarFieldType* dualNodalVolume_;
+  functor innerLoopFunctor_;
```

Comment:
`const size_t innerLoopExtent_;`
Add this here for populating on construction

Reply:
See my comment below